### DISCUSSION TITLE
Can listing page do pagination/partial query?

### DISCUSSION THREAD

**Discussion by NPhGit** [prometheus/pushgateway#579] [Q&A]
Can default pushgateway listing page do pagination/partial query?

**Comment by beorn7** [comment]
It cannot. However, if you have so many metrics pushed to the Pushgateway that this is a problem, you will have other problems, too.

Note that the PGW is not designed to turn Prometheus into a push-based monitoring system. It will struggle with more than a few hundreds of metrics.

If you want to push massive amount of metrics to a backend, you might want to check out the remote-write protocol (and a compatible backend designed to ingest a high throughput of metrics – Prometheus itself can act as a remote-write receiver, but isn't designed to use this at scale).
